pandas/core/indexes/interval.py

```diff
@@ -766,8 +766,13 @@ def get_loc(self, key, method=None):
                 key = Interval(left, right, key.closed)
             else:
                 key = self._maybe_cast_slice_bound(key, 'left', None)
-
-            start, stop = self._find_non_overlapping_monotonic_bounds(key)
+            try:
+                start, stop = self._find_non_overlapping_monotonic_bounds(key)
+            except TypeError:
+                # get loc should raise KeyError
+                # if key is hashable but
+                # of an incorrect type
+                raise KeyError
```

Comment:
This needs to be `raise from`, which is a Python 3 construct. To stay version-agnostic use `six.raise_from`.

```suggestion
import six
...
            try:
                start, stop = self._find_non_overlapping_monotonic_bounds(key)
            except TypeError as exc:
                six.raise_from(KeyError('Key is hashable, but of an incorrect type'), exc)
```

Reply:
OK, I've left the code as is but have improved the reported message as suggested by @rs2

Reply:
We don't have a dependency on six, so we can't use `six.raise_from`.

I'd recommend just dropping the original exception from the traceback and leaving `TODO(py3): use raise from`.